### ISSUE TITLE
Parser doesn't return all elements

### ISSUE THREAD

**Issue by tomasgarzon** [mjmlio/mjml#1145]
I think there is a bug in the parser. I have this code:

```xml
<mj-tag-list>
           <mj-tag>Keyword2</mj-tag>
           <mj-tag>Keyword2</mj-tag>
           <mj-tag>Keyword3</mj-tag>
           <mj-tag>Keyword10</mj-tag>
           <mj-tag>Keyword20</mj-tag>
           <mj-tag>Keyword30</mj-tag>
           <mj-tag>Keyword11</mj-tag>
           <mj-tag>Keyword21</mj-tag>
           <mj-tag>Keyword31</mj-tag>
       </mj-tag-list>
```
      
and this.props.children only have 8 elements, not the first one.

MjTagList:

```js
import { registerDependencies } from 'mjml-validator'
import { BodyComponent } from 'mjml-core'

registerDependencies({
  // Tell the validator which tags are allowed as our component's parent
  'mj-column': ['mj-tag-list'],
  // Tell the validator which tags are allowed as our component's children
  'mj-tag-list': []
})

/*
  Our component is a (useless) simple tag container.
*/
export default class MjTagList extends BodyComponent {

  // This functions allows to define styles that can be used when rendering (see render() below)
  getStyles() {
    return {
      wrapperDiv: {
        'width': "100%",
        'padding': '10px 25px'
      }
    }
  }

  renderTag() {
    const { children } = this.props

    return `
        ${this.renderChildren(children, {
          renderer: (
            component, // eslint-disable-line no-confusing-arrow
          ) =>
            component.constructor.isRawElement()
              ? component.render()
              : `
            ${component.render()}
          `,
        })}
    `
  }

  /*
    Render is the only required function in a component.
    It must return an html string.
  */
  render() {
    return `
      <div
        ${this.htmlAttributes({ // this.htmlAttributes() is the recommended way to pass attributes to html tags
          class: this.getAttribute('css-class'),
          style: 'wrapperDiv' // This will add the 'wrapperDiv' attributes from getStyles() as inline style
        })}
      >
          ${this.renderTag()}
      </div>
		`
  }
}
```

MjTag:
```js
import { registerDependencies } from 'mjml-validator'
import { BodyComponent } from 'mjml-core'

registerDependencies({
  // Tell the validator which tags are allowed as our component's parent
  'mj-tag-list': ['mj-tag'],
  // Tell the validator which tags are allowed as our component's children
  'mj-tag': []
})

export default class MjTag extends BodyComponent {
  // Tell the parser that our component won't contain other mjml tags
  static endingTag = true

  // Tells the validator which attributes are allowed for mj-layout
  static allowedAttributes = {
    'color': 'color',
    'font-size': 'unit(px)',
    'background-color': 'background-color'
  }

  // What the name suggests. Fallback value for this.getAttribute('attribute-name').
  static defaultAttributes = {
    color: '#37474F'
  }

  // This functions allows to define styles that can be used when rendering (see render() below)
  getStyles() {
    return {
      contentDiv: {
        'color': this.getAttribute('color'),
        'background-color': '#DEE1E2',
        'font-size': '14px',
        'font-weight': 'bold',
        'line-height': '21px',
        'border-radius': '50px',
        'text-align': 'center',
        'margin-right': '10px',
        'margin-bottom': '10px'
      }
    }
  }

  /*
    Render is the only required function in a component.
    It must return an html string.
  */
  render() {
    return `
      <div
        ${this.htmlAttributes({ // this.htmlAttributes() is the recommended way to pass attributes to html tags
          class: this.getAttribute('css-class'),
          style: 'contentDiv' // This will add the 'wrapperDiv' attributes from getStyles() as inline style
        })}
      >
        ${this.getContent()}
      </div>
		`
  }
}
```

**Reproduction Steps:**

1. Add these custom components
2. Create a <mj-tag-list> structure 
3. compile to html

**Expected behavior:**

We should have 9 elements

**Observed behavior:**

We only see 8 elements related to mj-tag

**MJML version:**

mjml-core: 4.0.1
mjml-cli: 4.0.3

**Email clients the bug is seen on:**

All html

**Comment by kmcb777** [comment]
@tomasgarzon Hi, thanks for reporting this. The fact that the parent has the same tag name as his child + a suffix messes up a regex at some point. The fix for this should be included in version 4.1.0

**Comment by kmcb777** [comment]
Hi @tomasgarzon we released a new beta for mjml 4.1, which you can install with `npm install mjml@next`, could you check if it resolves the problem you had ?
